sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -199,7 +199,7 @@ public boolean equals(ServiceSpec first, ServiceSpec second) {
         ServiceSpec loopbackSpecification = factory.parse(serviceSpec.getBytes());
         // Verify that equality works:
         if (!loopbackSpecification.equals(serviceSpec)) {
-            StringBuilder error = new StringBuilder();
+            StringBuilder error = new StringBuilder();  // TODO (arand) this is not a very helpful error message
```

Comment:
Tangentially related, but I hate basically everything about debugging spec validation errors. It'd be nice to fix that up some day.